packages/core/src/components/select/select.tsx

```diff
@@ -338,12 +409,17 @@ export class Select {
       return;
     }
 
+    let item;
+
     if (this.editable && !this.itemExists(this.inputFilterText)) {
       this.emitAddItem(this.inputFilterText);
-      this.navigationItem = this.items[this.items.length - 1];
+      item = this.items[this.items.length - 1];
+    }
+
+    if (item) {
+      item['onItemClick']?.();
```

Comment:
Define a function (annotate it with `@Method`) on the component and call it async.